src/stepfunctions/steps/states.py

```diff
@@ -701,7 +701,7 @@ def visit(self, state):
 
     def _validate_next_step_params(self, params, step_output):
         for k, v in params.items():
-            if isinstance(v, StepInput) and not step_output.contains(v):
+            if isinstance(v, StepInput) and v.name and not step_output.contains(v):
```

Comment:
Hi Andrew,

Thank you very much for the fix, we recommend the following updates to make the code more readable.

```python
            if isinstance(v, StepInput):
                if v is not step_output and v not in step_output:
                    return False, k
```

This requires a change in the [placeholders.py](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/master/src/stepfunctions/inputs/placeholders.py#L134) with the following modifications.

```python
    def __contains__(self, placeholder):
        return self.contains(placeholder)
```

Could you submit another revision, we will be very happy to merge it.

Best,
Shunjia

Reply:
Thanks

Re: the first change. Minor - this does change the logic slightly i.e. it will no longer potentially enter the elif if `step_output.contains(v)` is true. This is probably fine and the tests pass cleanly.

The second suggested change causes infinite recursion in a number of tests
```
src/stepfunctions/inputs/placeholders.py:128: in contains
    return self.contains(placeholder)
src/stepfunctions/inputs/placeholders.py:128: in contains
    return self.contains(placeholder)
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)
```
Please let me know how you want to want to proceed.

Andy

Reply:
For the second part, we have tested it locally here, it worked. Could you make a commit so that we can review the change?

Shunjia